AppCenter/AppCenterTests/MSCSExtensionsTests.m

```diff
@@ -765,14 +765,20 @@ - (void)testDeviceExtIsEqual {
 
 #pragma mark - MSCSData
 
-- (void)testDataJSONSerializingToDictionary {
+- (void)testDataJSONSerializingToDictionaryIsOrdered {
 
   // When
   NSMutableDictionary *dict = [self.data serializeToDictionary];
 
   // Then
-  XCTAssertNotNil(dict);
-  XCTAssertEqualObjects(dict, self.dataDummyValues);
+  XCTAssertNotNil([self.data serializeToDictionary]);
+
+  // Only verify the order for baseType and baseData fields.
+  XCTAssertTrue([[dict allKeys][0] isEqualToString:@"baseType"]);
+  XCTAssertTrue([[dict allKeys][1] isEqualToString:@"baseData.someData"]);
+  XCTAssertEqualObjects(dict[@"aKey"], @"aValue");
```

Comment:
I wonder if there is a way to extract dictionaries exempt of "baseData.someData" and "baseType" then use them for the comparison :thinking:. That way we don't have to hardcode the other entries.

Reply:
Nevermind it adds too much complexity for just that one